src/aaudio/AAudioExtensions.h

```diff
@@ -80,12 +144,27 @@ class AAudioExtensions {
      * @return 0 or a negative error code
      */
     int32_t setMMapEnabled(bool enabled) {
+        // The API for setting mmap policy is on public after API level 36.
+        if (mLibLoader->aaudio_setMMapPolicy != nullptr) {
+            return mLibLoader->aaudio_setMMapPolicy(
+                    static_cast<aaudio_policy_t>(enabled ? MMapPolicy::Auto : MMapPolicy::Never));
+        }
+        // When there is no public API, fallback to loading the symbol from hidden API.
         if (loadSymbols()) return AAUDIO_ERROR_UNAVAILABLE;
         if (mAAudio_setMMapPolicy == nullptr) return false;
-        return mAAudio_setMMapPolicy(enabled ? AAUDIO_POLICY_AUTO : AAUDIO_POLICY_NEVER);
+        return mAAudio_setMMapPolicy(
+                static_cast<int32_t>(enabled ? MMapPolicy::Auto : MMapPolicy::Never));
     }
 
     bool isMMapEnabled() {
+        // The API for getting mmap policy is on public after API level 36.
+        // Use it when it is available.
+        if (mLibLoader->aaudio_getMMapPolicy != nullptr) {
```

Comment:
The loadsymbols() call is after this check.
And loadsymbols() sets mLibLoader->aaudio_getMMapPolicy.
So I think the first time this gets called then it will fall back to using mAAudio_getMMapPolicy().
The second call will find the symbol and use it.
Maybe loadsymbols() should come first.

But I may be tracing the code wrong...

Reply:
Oh, that is a good point! I should also modify `loadSymbols`. The LibLoader will be opened from the constructor of AAudioExtensions when calling `initMMapPolicy`. If it is opened successfully, `mLibLoader->aaudio_getMMapPolicy` will be set if it is on Android 16+. Otherwise, it fallbacks to the old logic.

Reply:
> The LibLoader will be opened from the constructor of AAudioExtensions when calling initMMapPolicy

In the new code:
AAudioExtensions() constructor calls initMMapPolicy():

```
AAudioExtensions() {
        mLibLoader = AAudioLoader::getInstance();
        if (!initMMapPolicy()) {
```

Since mLibLoader is non-null, the open() will get called:

```
    bool initMMapPolicy() {
        if (mLibLoader == nullptr || mLibLoader->open() != 0) {
            return false;
        }
```

The open() method sets aaudio_getPlatformMMapPolicy and aaudio_getMMapPolicy.

So I guess this is OK.